src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -274,12 +285,28 @@ public bool TryWriteBytes(Span<byte> destination, out int bytesWritten)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void WriteIPv6Bytes(Span<byte> destination)
         {
-            Debug.Assert(_numbers != null && _numbers.Length == NumberOfLabels);
-            int j = 0;
-            for (int i = 0; i < NumberOfLabels; i++)
+            ushort[]? numbers = _numbers;
+            Debug.Assert(numbers != null && numbers.Length == NumberOfLabels);
+
+            if (BitConverter.IsLittleEndian)
+            {
+                if (Vector128.IsHardwareAccelerated)
+                {
+                    Vector128<ushort> ushorts = Vector128.LoadUnsafe(ref MemoryMarshal.GetArrayDataReference(numbers));
+                    ushorts = Vector128.ShiftLeft(ushorts, 8) | Vector128.ShiftRightLogical(ushorts, 8);
+                    ushorts.AsByte().StoreUnsafe(ref MemoryMarshal.GetReference(destination));
```

Comment:
Should we should define a `Vector128.ReverseEndianness` API for the supported integer types and then call that instead?

We can then open an API proposal and take it to review. It seems like a common enough scenario that having a dedicated API would be beneficial and would allow specifics of things like `shift` vs `shuffle` vs `...` being optimized internally by the JIT.

Reply:
If you think such an API makes sense there, I have no objections.

I also opened https://github.com/dotnet/runtime/issues/75901.